src/sql/src/plan/statement/ddl.rs

```diff
@@ -5103,10 +5103,21 @@ pub fn describe_alter_role(
 
 pub fn plan_alter_role(
     _: &StatementContext,
-    AlterRoleStatement { name, options }: AlterRoleStatement<Aug>,
+    AlterRoleStatement {
+        name,
+        options,
+        variable,
+    }: AlterRoleStatement<Aug>,
 ) -> Result<Plan, PlanError> {
     let attributes = plan_role_attributes(options)?;
 
+    if variable.is_some() {
```

Comment:
According to the postgres docs (https://www.postgresql.org/docs/15/sql-alterrole.html), both options and variable cannot be simultaneously present. An error path could be added for that already here. Other option is to produce a new AST variant so the parser takes care of that enforcement.

Reply:
Either works for me! Do you have an opinion on which would be better? Not sure if we have a rule of thumb for this kind of thing

Reply:
I'd favor a parse error because postgres describes these as separate statement variants, so it would be reasonable to assume it's not even possible to construct an invalid one.

Reply:
From the parser I think it's impossible to do both, since we support only one of `WITH`, `SET`, or `RESET` at a time. I updated #21881 to make the statement and planning data types reflect that only one thing can occur at a time.